Dz4

aagaryov74 opened the pull request.

Commit:
rewrite with spring shell

Commit:
reformat code

Commit:
refactoring for last updates from dz3

Commit:
refactoring CsvFileReader because Localizator

Commit:
refactoring - creating YamlMessagesSource

Commit:
change banner and settings

Commit:
Александр Оруджев17.03.2019 в 21:55
Ну да. Почти то что я и имел в виду. Туда бы еще работу с MessageSource вынести и то что получилось использовать в IOService. Сам MessageSource можно завернуть в YamlMessageSource. Давайте тогда эту работу я принимаю. Если захотите еще поправить то можно написать сюда или поправить в следующей работе. Только напишите тогда там об том что исправили замечания из прошлого ДЗ.
17.03.2019 в 21:55 Александр Оруджев принял ДЗ
Алексей Агарёв17.03.2019 в 21:58
Спасибо! Я тогда в дз на shell это доделаю

Commit:
fix tests and delete unused methods

stvort commented on src/main/java/ru/otus/agaryov/dz4/service/IOServiceImpl.java:
```diff
@@ -0,0 +1,47 @@
+@Service
+public class IOServiceImpl implements  IOService {
+    private final MessageSource messageSource;
+    private Locale locale;
```
Зачем IOServiceImpl своя локаль если он ее использует для вызова getMessage YamlMessageSource, у которой тоже есть своя локаль? Давайте оставим только в YamlMessageSource. Заодно getMessage  сократим

aagaryov74 replied:
Да я думал над этим, но потом подумал что и так мы все перетащили в локализатор - если еще и локаль оттуда вытащить то он как обертка над printf  вообще смысл потеряет и оставил локаль.

aagaryov74 replied:
https://github.com/aagaryov74/otus-home-work4/pull/1/commits/e2d501aabbae36e279b69edc7c660e59cc331cfd

aagaryov74 replied:
Переделал

stvort commented on src/main/java/ru/otus/agaryov/dz4/exam/ExamExecutorImpl.java:
```diff
@@ -0,0 +1,88 @@
+                ioService.printToConsole("ioWarning");
+            }
+        } else {
+            ioService.printToConsole("sayhellofirst");
```
Порядок команд лучше обеспечить с помощью ```@ShellMethodAvailability```

aagaryov74 replied:
Спасибо!
Исправлю

aagaryov74 replied:
https://github.com/aagaryov74/otus-home-work4/pull/1/commits/c6031bc2835ede66c031450559bf26bd11d0f14e

stvort commented on src/main/java/ru/otus/agaryov/dz4/exam/ExamExecutorImpl.java:
```diff
@@ -0,0 +1,88 @@
+    @Override
+    public void doExam() {
+        doexam();
+    }
```
Зачем два doExam? Возможно лучше воспользоваться параметром key аннотации ```@ShellMethod```

aagaryov74 replied:
Как только ставишь аннотацию @ShellMethod у doExam в имплементации у меня Идея сразу говорит что метод из интерфейса не имплементирован.  Key - это не то, это набор алиасов как я понял по которым эта команда будет из шелла доступна

Commit:
stvort 10 hours ago
Зачем IOServiceImpl своя локаль если он ее использует для вызова getMessage YamlMessageSource, у которой тоже есть своя локаль? Давайте оставим только в YamlMessageSource. Заодно getMessage сократим

Commit:
stvort 10 hours ago
Порядок команд лучше обеспечить с помощью @ShellMethodAvailability

Commit:
stvort 21 hours ago
Зачем два doExam? Возможно лучше воспользоваться параметром key аннотации @ShellMethod
